moveit_ros/planning/rdf_loader/src/rdf_loader.cpp

```diff
@@ -49,48 +51,78 @@
 #include <fstream>
 #include <streambuf>
 #include <algorithm>
+#include <chrono>
 
-rdf_loader::RDFLoader::RDFLoader(const std::string& robot_description)
+rclcpp::Logger LOGGER_RDF_LOADER = rclcpp::get_logger("moveit").get_child("rdf_loader");
+
+rdf_loader::RDFLoader::RDFLoader(const std::shared_ptr<rclcpp::Node>& node, const std::string& robot_description)
 {
   moveit::tools::Profiler::ScopedStart prof_start;
   moveit::tools::Profiler::ScopedBlock prof_block("RDFLoader(robot_description)");
 
-  ros::WallTime start = ros::WallTime::now();
-  ros::NodeHandle nh("~");
-  std::string content;
+  auto start = std::chrono::system_clock::now();
 
-  if (!nh.searchParam(robot_description, robot_description_) || !nh.getParam(robot_description_, content))
+  std::string content;
+  // TODO(JafarAbdi): Revise parameter lookup
+  if (node->has_parameter(robot_description))
+  {
+    rclcpp::Parameter robot_description_param = node->get_parameter(robot_description);
+    try
+    {
+      content = robot_description_param.as_string();
+    }
+    catch (const rclcpp::ParameterTypeException& e)
+    {
+      RCLCPP_WARN(LOGGER_RDF_LOADER, "When getting robot_description parameter %s", e.what());
+    }
+  }
+  if (content.length() < 1)
   {
-    ROS_ERROR_NAMED("rdf_loader", "Robot model parameter not found! Did you remap '%s'?", robot_description.c_str());
+    RCLCPP_INFO_ONCE(LOGGER_RDF_LOADER, "Waiting for Robot model topic! Did you remap '%s'?\n",
```

Comment:
This message does not belong here, it still references the topic from the workaround